source/merge.ts

```diff
@@ -0,0 +1,85 @@
+import {URL} from 'url';
+import is from '@sindresorhus/is';
+import {Options, Method} from './utils/types';
+import knownHookEvents, {Hooks, HookType, HookEvent} from './known-hook-events';
+
+export default function merge(target: Options, ...sources: Options[]) {
+	for (const source of sources) {
+		for (const [key, sourceValue] of Object.entries(source)) {
+			if (is.undefined(sourceValue)) {
+				continue;
+			}
+
+			const targetValue = target[key];
+			if (is.urlInstance(targetValue) && (is.urlInstance(sourceValue) || is.string(sourceValue))) {
+				target[key] = new URL(sourceValue as string, targetValue);
+			} else if (is.plainObject(sourceValue)) {
+				if (is.plainObject(targetValue)) {
+					target[key] = merge({}, targetValue as Options, sourceValue as Options);
+				} else {
+					target[key] = merge({}, sourceValue as Options);
+				}
+			} else if (is.array(sourceValue)) {
+				target[key] = merge([], sourceValue);
+			} else {
+				target[key] = sourceValue;
+			}
+		}
+	}
+
+	return target;
+}
+
+export function mergeOptions(...sources: Options[]) {
+	sources = sources.map(source => source || {});
+	const merged = merge({}, ...sources);
```

Comment:
The type of `merged` should be `Options`

Reply:
Added explicit return type for `merged` function.